assignmentPR/MathFunc1.c

```diff
@@ -0,0 +1,316 @@
+        }
+    }
+    return *(unsigned char *)
+                       stringOne < *(unsigned char *)stringTwo
+               ? -1
+               : 1;
+}
+
+// "clean out" a given char array by filling it entirely with the null char.
+// this prevents the presence of any "UN-Readable" character within this array
+void cleanCharArray(char *charArray, int arraySize)
+{
+    int index;
+    for (index = 0; index < arraySize; index++)
+    {
+        charArray[index] = '\0';
+    }
+}
+
+
+// Modified from https://www.geeksforgeeks.org/write-your-own-atoi/
+// A utility function to check whether character is numeric
+int isNumericChar(char character)
+{
+    return (character >= '0' && character <= '9' || character == '-') ? true : false;
+}
+
+
+// Modified from https://www.geeksforgeeks.org/write-your-own-atoi/
+// A simple atoi() function. If the given string contains
+// any invalid character, then this function returns INT_CONVERSION_ERROR
+int myAtoi(char *stringSequence)
+{
+    int negativeFlag = 0;
+
+    if (compareString(stringSequence, "\0") == SAME_STRING)
+    {
+        return INT_CONVERSION_ERROR;
+    }
+
+    int result = 0; // Initialize result
+    int index = 0;  // Initialize index of first digit
+
+    // If number is negative, then fail gracefully
+    if (stringSequence[0] == '-')
+    {
+        negativeFlag = 1;
+        index = 1;
+    }
+
+    // Iterate through all digits of input string and update result
+    for (; stringSequence[index] != '\0'; ++index)
+    {
+        if (isNumericChar(stringSequence[index]) == false)
+        {
+            if ((int)(stringSequence[index] <= WHITESPACE))
+            {
+                continue;
+            }
+            else if (stringSequence[index] == '.')
+            {
+                break;
+            }
+
+            else
+            {
+                return INT_CONVERSION_ERROR;
+            }
+        }
+        else
+        {
+            result = (result * 10) + stringSequence[index] - '0';
+        }
+    }
+
+    if ( negativeFlag == 1) 
+    {
+        return -1 *result;
+    }
+    return result;
+}
+
+
+/* Modified from https://en.wikibooks.org/wiki/C_Programming/stdlib.h/itoa */
+void reverse(char string[])
+{
+    int count, stringLength;
+    char character;
+
+    for (count = 0, stringLength = myStrLen(string) - 1; count < stringLength;
+         count++, stringLength--)
+    {
+        character = string[count];
+        string[count] = string[stringLength];
+        string[stringLength] = character;
+    }
+}
+
+
+/* Returns the length of a char array */
+int myStrLen(char *string)
+{
+    int count = 0;
+    while (string[count] != '\0')
+    {
+        count++;
+    }
+    return count;
+}
+
+
+/* Returns the numeric value of 10 ^ stringLength */
+int tenToThePowerOf(int stringLength)
+{
+    int answer = 1;
+    int iterator;
+    for (iterator = 0; iterator < stringLength; iterator++)
+    {
+        answer *= 10;
+    }
+    return answer;
+}
```

Comment:
No newline at EOF....................................... (lol)

Reply:
Sadly with ATOM every time I save it forces a newline at the end of the file!